graphql/src/Mu/GraphQL/Quasi.hs

```diff
@@ -0,0 +1,187 @@
+{-# language DataKinds         #-}
+{-# language OverloadedStrings #-}
+{-# language TemplateHaskell   #-}
+{-# language ViewPatterns      #-}
+
+module Mu.GraphQL.Quasi (
+  -- * 'Schema' generation from @.graphql@ files
+  graphql
+) where
+
+import           Control.Monad.IO.Class        (liftIO)
+import           Data.Coerce                   (coerce)
+import qualified Data.HashMap.Strict           as HM
+import qualified Data.Text                     as T
+import qualified Data.Text.IO                  as TIO
+import           Data.UUID                     (UUID)
+import qualified Language.GraphQL.Draft.Syntax as GQL
+import           Language.Haskell.TH
+
+import           Mu.GraphQL.Annotations
+import           Mu.GraphQL.Quasi.LostParser   (parseTypeSysDefinition)
+import           Mu.Rpc
+import           Mu.Schema.Definition
+
+-- | Imports an graphql definition written in-line as a 'Schema'.
+graphql :: String -> String -> FilePath -> Q [Dec]
+graphql scName svName file = do
+  schema <- liftIO $ TIO.readFile file
+  case parseTypeSysDefinition schema of
+    Left e  -> fail ("could not parse graphql spec: " ++ show e)
+    Right p -> graphqlToDecls scName svName p
+
+type TypeMap   = HM.HashMap T.Text GQLType
+type SchemaMap = HM.HashMap T.Text GQL.OperationType
+
+data Result =
+    GQLScalar
+  | GQLSchema Type
+  | GQLService Type
+
+data GQLType =
+    Enum
+  | Object
+  | Scalar
+  | InputObject
+  | Other
+
+classifySchema :: [GQL.SchemaDefinition] -> SchemaMap
+classifySchema = HM.fromList . concatMap schemaToMap
```

Comment:
The code is OK, but we aware that unless GHC does extreme optimization, you are here generating an intermediate list which is them discarded. I would suggest to use `foldl'`, using the `HashMap`'s `insert` at each step. On the other hand, now you can use `concatMap`, which is quite useful...